### ISSUE TITLE
Feature request: soft line wrapping

### ISSUE THREAD

**Issue by jozsefk9** [neurocyte/flow#89]
It would be very useful to have soft line wrap available as an option in flow editor. Please consider implementing this feature. Thank you.

**Comment by tamadamas** [comment]
@neurocyte @jozsefk9 I'm going to implement it as it crusual for my usage of editor just now.

So, I think it relates with #208 cause soft wrap will on one of viewpoint width or first configured rule number.
As I remember it usually works in editors I use, right?

**Comment by neurocyte** [comment]
> [@neurocyte](https://github.com/neurocyte) [@jozsefk9](https://github.com/jozsefk9) I'm going to implement it as it crusual for my daily with the editor.
> 
> I believe this issue is related to [#208](https://github.com/neurocyte/flow/issues/208), as soft wrap seems to occur based on either the viewport width or the first configured rule number. In my experience, this behavior is standard in most code editors I've used. Could we confirm if this is the expected functionality?

I would expect hard re-wrap commands (if they existed) to respect the vertical rulers and soft wrap to use the view width. That being said, I think it'd be easy to make this configurable.

**Comment by jozsefk9** [comment]
> @neurocyte @jozsefk9 I'm going to implement it as it crusual for my daily with the editor.
> 
> I believe this issue is related to #208, as soft wrap seems to occur based on either the viewport width or the first configured rule number. In my experience, this behavior is standard in most code editors I've used. Could we confirm if this is the expected functionality?

Yes, please. That's the expected functionality. Thank you for helping!

**Comment by neurocyte** [comment]
> I'm going to implement it as it crusual for my daily with the editor.

This is going to be a pretty big task. I expect soft wrap support to require major additions/changes to the `Buffer` module, and also big changes to the renderer in the editor and the gutter, and also to navigation related commands

The first step is to add support for soft `bols` and soft `eols` to the `Weights` structures in the `Buffer` and then modify the buffer loader to calculate/split nodes into soft wrap width sized chunks.

Then comes the huge task of figuring out when to use `bols`/`eols` and when to use `soft_bols`/`soft_eols`, etc.

**Comment by tamadamas** [comment]
> > [@neurocyte](https://github.com/neurocyte) [@jozsefk9](https://github.com/jozsefk9) I'm going to implement it as it crusual for my daily with the editor.
> > I believe this issue is related to [#208](https://github.com/neurocyte/flow/issues/208), as soft wrap seems to occur based on either the viewport width or the first configured rule number. In my experience, this behavior is standard in most code editors I've used. Could we confirm if this is the expected functionality? > 
> I would expect hard re-wrap commands (if they existed) to respect the vertical rulers and soft wrap to use the view width. That being said, I think it'd be easy to make this configurable.

Ah, sure, I definitely add config for that thing. And also I check how others do that. We don't want to reinvent a wheel on such simple task

**Comment by tamadamas** [comment]
> > I'm going to implement it as it crusual for my daily with the editor.
> 
> This is going to be a pretty big task. I expect soft wrap support to require major additions/changes to the `Buffer` module, and also big changes to the renderer in the editor and the gutter, and also to navigation related commands
> 
> The first step is to add support for soft `bols` and soft `eols` to the `Weights` structures in the `Buffer` and then modify the buffer loader to calculate/split nodes into soft wrap width sized chunks.
> 
> Then comes the huge task of figuring out when to use `bols`/`eols` and when to use `soft_bols`/`soft_eols`, etc.
Thanks, I'll check and come with questions. Is Discord active? I could try there if it's better

**Comment by neurocyte** [comment]
Yes, discord is the place to be.

**Comment by plutonium-239** [comment]
is this still planned?
